scripts/xtensa-build-zephyr.py

```diff
@@ -323,6 +326,25 @@ def west_init_update():
 	# Do NOT "west update sof"!!
 	execute_command(["west", "update", "zephyr", "hal_xtensa"], timeout=300, cwd=west_top)
 
+def get_sof_version(abs_build_dir):
+	"""[summary] Get version string major.minor.micro of SOF firmware file.
+	When building multiple platforms from the same SOF commit, all platforms share the same version.
+	So for the 1st platform, generate the version string from sof_version.h and later plaforms will
+	reuse it.
+	"""
+	global sof_version
```

Comment:
The feature can be done by:
- use a global sof_version, and modify it with `get_sof_version`. in signing, use the global one directly.
- don't use a global sof_version, and in signing, use the one returned from `get_sof_version`

Seems you mixed use above two in the code, I prefer the second one without the global version. so the code of `get_sof_version` can be simplified as:

```

versions = {}
with open(pathlib.Path(abs_build_dir, "zephyr/include/generated/sof_versions.h")) as hfile:
		for hline in hfile:
			words = hline.split()
			if words[0] == '#define':
				versions[words[1]] = words[2]
return versions['SOF_MAJOR'] + '.' + versions['SOF_MINOR'] + '.' + versions['SOF_MICRO']
```

Reply:
I agree with @aiChaoSONG that there is a problem now: the `sof_version` global is set twice, BOTH inside and outside the function. That's not good.

However I prefer the other option: keep all read/write accesses to the global `sof_version` and its "maintenance" _private_ and isolated inside the function.

Reply:
Oh wait: `sof_version` is local, shadowing the global. That's not good either; shadowing == bad.
```
pylint scripts/xtensa-build-zephyr.py| sort -t: -k2,3 -n

scripts/xtensa-build-zephyr.py:472:2: W0621: Redefining name 'sof_version' from outer scope (line 27) (redefined-outer-name)

```

https://pylint.pycqa.org/en/latest/user_guide/messages/warning/redefined-outer-name.html

Same solution.